ibis/backends/duckdb/__init__.py

```diff
@@ -70,6 +71,14 @@ class Backend(BaseAlchemyBackend):
     def current_database(self) -> str:
         return "main"
 
+    @contextlib.contextmanager
+    def begin(self):
+        with super().begin() as con:
+            tz = con.execute(sa.select(sa.func.current_setting("TimeZone"))).scalar()
+            con.exec_driver_sql("SET TimeZone = 'UTC'")
+            yield con
+            con.exec_driver_sql(f"SET TimeZone = '{tz}'")
```

Comment:
Can you add a comment here explaining what's being done and why by this block?

Alternatively, are these settings persisted, or are they just for a connection? If so, could we instead set the timezone to UTC on connect and not worry about it here?

Reply:
Looks like they are persisted for a connection/session and not stored in the db so I think we can do this in a `connect` event.

Reply:
This is probably true of all the sqlalchemy backends, so I can look into do that more broadly which would cut down on the number of things we're executing and prevent having to do the cleanup all the time.

Reply:
Fixed for duckdb.